### DISCUSSION TITLE
Docker Installation (Windows)

### DISCUSSION THREAD

**Discussion by sngm** [pimcore/pimcore#16209] [Q&A / Support] [answered]
Hello.

I follow this: https://pimcore.com/docs/platform/Pimcore/Getting_Started/Installation/Docker_Based_Installation/

I run the first command from Step 1:

`# demo package with exemplary blueprints (`pimcore/demo`)
docker run -u `id -u`:`id -g` --rm -v `pwd`:/var/www/html pimcore/pimcore:php8.2-latest composer create-project pimcore/demo my-project`

The command installed all the symfony stuff, so I think it was successfully. The last lines in my terminal were:

```
> Pimcore\Composer::installAssets

 Trying to install assets as relative symbolic links.

 --- ------------------------------------------ ------------------
      Bundle                                     Method / Error
 --- ------------------------------------------ ------------------
  ✔   FOSJsRoutingBundle                         relative symlink
  ✔   PimcoreAdminBundle                         relative symlink
  ✔   PimcoreCustomReportsBundle                 relative symlink
  ✔   PimcoreEcommerceFrameworkBundle            relative symlink
  ✔   PimcoreApplicationLoggerBundle             relative symlink
  ✔   PimcorePersonalizationBundle               relative symlink
  ✔   PimcoreGoogleMarketingBundle               relative symlink
  ✔   PimcoreCustomerManagementFrameworkBundle   relative symlink
  ✔   PimcoreNewsletterBundle                    relative symlink
  ✔   PimcoreGlossaryBundle                      relative symlink
  ✔   PimcoreSeoBundle                           relative symlink
  ✔   PimcoreSimpleBackendSearchBundle           relative symlink
  ✔   PimcoreDataHubBundle                       relative symlink
  ✔   OutputDataConfigToolkitBundle              relative symlink
  ✔   Web2PrintToolsBundle                       relative symlink
  ✔   ObjectMergerBundle                         relative symlink
  ✔   PimcoreWebToPrintBundle                    relative symlink
  ✔   PimcoreTinymceBundle                       relative symlink
  ✔   PimcoreStaticRoutesBundle                  relative symlink
  ✔   PimcoreWordExportBundle                    relative symlink
  ✔   PimcoreXliffBundle                         relative symlink
  ✔   PimcoreCoreBundle                          relative symlink
 --- ------------------------------------------ ------------------

 [OK] All assets were successfully installed.

No security vulnerability advisories found.
> Pimcore\Composer::postCreateProject
```

Under Step 2 they say "Go to your new project cd my-project/" .... but where is this folder "my-project"? I can't find it.

Where I have to look for the folder and the docker compose file? 

Thanks for any advise for this topic!

Greetings Magnus

**Comment by kingjia90** [comment]
Wherever you ran the command, it would be in a folder under that path, if you type `dir` in the terminal, you should see it if you didn't navigate (via `cd` or `drive`) to any other location

**Comment by martenw** [comment]
I am having the same problem, did you find any solution @sngm?

Can also add that when I tried to run  the `docker run` command from [documentation](https://pimcore.com/docs/platform/Pimcore/Getting_Started/Installation/Docker_Based_Installation) in PowerShell I got an error:

```PowerShell
> docker run -u `id -u`:`id -g` --rm -v `pwd`:/var/www/html pimcore/pimcore:php8.2-latest composer create-project pimcore/demo my-project
unknown shorthand flag: 'g' in -g`
```

Adjusted to the command below which ran the installation without any error at least, but no folder got created:

```PowerShell
> docker run -u 1000:1000 --rm -v ./var/www/html pimcore/pimcore:php8.2-latest composer create-project pimcore/demo my-project
```

**Reply by sngm** [reply]
@martenw 

I have now restarted again and it has actually worked.

I am still using Windows 10 + WSL2 + Docker

I use WSL2 Ubuntu. In this terminal I have also executed all commands from the installation page "https://pimcore.com/docs/platform/Pimcore/Getting_Started/Installation/Docker_Based_Installation".

This worked directly without any problems. Have you been able to solve it in the meantime?

I used the demo installation and also commented the line "user" in the docker-composer.

The project folder is located directly in the wsl file system, where I executed the following command:

```
docker run -u `id -u`:`id -g` --rm -v `pwd`:/var/www/html pimcore/pimcore:php8.2-latest composer create-project pimcore/demo my-project
```
 I have done this under /home/docker/pimcore/. And my project folder is now under /home/docker/pimcore/my-project

Currently I use the VS Code extension "WSL" to connect directly to the file system under WSL and to be able to open and edit the folder in VS Code. I find this very practical.

**Reply by martenw** [reply, marked as answer]
I also switched to running inside WSL2 and it works fine now. I wrote down some notes for myself in the process that might be useful for someone else also: https://gist.github.com/martenw/b7a64ebf6155276df9922fb343999609
